### PR TITLE
KAFKA-7992: Introduce start-time-ms metric

### PR DESCRIPTION
**Pull request by stanislavkozlovski**
[KIP-436](https://cwiki.apache.org/confluence/display/KAFKA/KIP-436%3A+Add+a+metric+indicating+start+time)
[KAFKA-7992](https://issues.apache.org/jira/browse/KAFKA-7992)

This PR exposes a new `start-time-ms` metric aross all components

Also removes the unused scala `AppInfo` class

### CODE REVIEW

File: core/src/main/scala/kafka/common/AppInfo.scala [hunk 1]
```diff
@@ -1,54 +0,0 @@
...
- * limitations under the License.
- */
-
-package kafka.common
-
-import com.yammer.metrics.core.Gauge
-import kafka.metrics.KafkaMetricsGroup
-import org.apache.kafka.common.utils.AppInfoParser
-
-object AppInfo extends KafkaMetricsGroup {
```
Comment (ijuma) [inline core/src/main/scala/kafka/common/AppInfo.scala:24]: Is this unused?

Comment (ijuma) [reply core/src/main/scala/kafka/common/AppInfo.scala:24]: You said in the PR description that it is unused, nevermind.

**Comment by ijuma** [comment]
Thanks for the PR. Are you planning to add tests?

**Comment by stanislavkozlovski** [comment]
@ijuma I wasn't planning to but thanks for bringing that up - it's definitely a good idea to have some. I added two tests for `AppInfoParser`, mainly about verifying it registers/unregisters metrics

**Comment by cmccabe** [comment]
LGTM.  Thanks for adding the test

**Comment by cmccabe** [comment]
Retest this please

**Comment by stanislavkozlovski** [comment]
Retest this please

**Pull request merged**